[FX] Fix GraphModule copy methods not regenerating forward

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #44830 [FX] Make Graphs immutable after output is set
* **#44806 [FX] Fix GraphModule copy methods not regenerating forward**

Differential Revision: [D23738732](https://our.internmc.facebook.com/intern/diff/D23738732)

Commit:
[FX] Fix GraphModule copy methods not regenerating forward

[ghstack-poisoned]

zdevito commented on torch/fx/graph_module.py:
```diff
@@ -124,10 +124,12 @@ def __reduce__(self):
-        the_copy.__dict__ = deepcopy(self.__dict__, memo)
-        return the_copy
+        fake_mod = torch.nn.Module()
+        fake_mod._modules = copy.deepcopy(self._modules)
```
This needs to copy all the properties in `self`, it misses attributes, parameters, buffers, etc.

zdevito replied:
```
m = torch.nn.Module()
m.__dict__ = copy.deepcopy(self.__dict__)
return GraphModule(m, self.graph)
```

Commit:
Update on "[FX] Fix GraphModule copy methods not regenerating forward"


Differential Revision: [D23738732](https://our.internmc.facebook.com/intern/diff/D23738732)

[ghstack-poisoned]

Commit:
Update on "[FX] Fix GraphModule copy methods not regenerating forward"


Differential Revision: [D23738732](https://our.internmc.facebook.com/intern/diff/D23738732)

[ghstack-poisoned]
